src/python/python/ad/largesteps.py

```diff
@@ -0,0 +1,175 @@
+from __future__ import annotations as __annotations__ # Delayed parsing of type annotations
+
+import mitsuba as mi
+import drjit as dr
+import numpy as np
+
+def mesh_laplacian(n_verts, faces, lambda_):
+    """
+    Compute the index and data arrays of the (combinatorial) Laplacian matrix of
+    a given mesh.
+    """
+    # Neighbor indices
+    ii = faces[:, [1, 2, 0]].flatten()
+    jj = faces[:, [2, 0, 1]].flatten()
+    adj = np.unique(np.stack([np.concatenate([ii, jj]), np.concatenate([jj, ii])], axis=0), axis=1)
+    adj_values = np.ones(adj.shape[1], dtype=np.float64) * lambda_
+
+    # Diagonal indices, duplicated as many times as the connectivity of each index
+    diag_idx = np.stack((adj[0], adj[0]), axis=0)
+
+    diag = np.stack((np.arange(n_verts), np.arange(n_verts)), axis=0)
+
+    # Build the sparse matrix
+    idx = np.concatenate((adj, diag_idx, diag), axis=1)
+    values = np.concatenate((-adj_values, adj_values, np.ones(n_verts)))
+
+    return idx, values
+
+class SolveCholesky(dr.CustomOp):
+    """
+    DrJIT custom operator to solve a linear system using a Cholesky factorization.
+    """
+
+    def eval(self, solver, u):
+        self.solver = solver
+        x = dr.empty(dr.cuda.TensorXf, shape=u.shape)
+        solver.solve(u, x)
+        return mi.TensorXf(x)
+
+    def forward(self):
+        x = dr.empty(mi.TensorXf, shape=self.grad_in('u').shape)
+        self.solver.solve(self.grad_in('u'), x)
+        self.set_grad_out(x)
+
+    def backward(self):
+        x = dr.empty(dr.cuda.TensorXf, shape=self.grad_out().shape)
```

Comment:
Changing the type to `mi.TensorXf` allows this to work in llvm modes as well i.e.
```python
        x = dr.empty(mi.TensorXf, shape=self.grad_out().shape)

```